yarpc/src/http/method.rs

```diff
@@ -0,0 +1,42 @@
+use std::collections::HashMap;
+
+use super::{message::PROTOCOL, Request};
+
+#[derive(Debug, PartialEq, Clone, Copy, Eq)]
+pub enum Method {
+    GET,
+    POST,
+}
+
+const GET: &str = "GET";
+const POST: &str = "POST";
+
+impl Method {
+    pub const fn to_str(self) -> &'static str {
+        match self {
+            Method::GET => GET,
+            Method::POST => POST,
+        }
+    }
```

Comment:
This method does not need to take ownership of self
```suggestion
    pub const fn as_str(&self) -> &'static str {
        match self {
            Method::GET => GET,
            Method::POST => POST,
        }
    }
```

Reply:
In addition, I suggest implementing `Display` for this type.

Reply:
This enum is cloneable, copyable ant it's better to pass by value (like u8).

Reply:
Fair enough.